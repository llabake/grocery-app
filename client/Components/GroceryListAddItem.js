import React, { Component } from 'react';
import { connect } from "react-redux";
import { addGrocery } from "../actions/groceryItemAction";


export class GroceryListAddItem extends Component {
  state = {
    item: {
      name: '',
    },
  };

  handleInputChange = (event) => {
    this.setState({
      item: { name: event.target.value },
    })
  };

  handleAddItem = (event) => {
    event.preventDefault();
    this.props.addGrocery(this.state.item)
      .then(() => this.setState(
        { item: { name: '' } },
      ))
  };

  render() {
    return (
      <div className='grocery-addItem'>
        <form onSubmit={this.handleAddItem}>
          <input
            value={this.state.item.name}
            onChange={this.handleInputChange}
          />
          <button type="submit">Add Item</button>
        </form>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  addGrocery: grocery => dispatch(addGrocery(grocery)),
});

export default connect(null, mapDispatchToProps)(GroceryListAddItem)