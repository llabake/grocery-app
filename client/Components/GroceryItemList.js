import React from 'react';

import GroceryListAddItem from "./GroceryListAddItem";
import GroceryItem from "./GroceryItem";

const GroceryItemList = ({ items }) => (
  <div>
    <h1 className='title'>Grocery List</h1>
    <div>
      {
        items.map(item => (
          <GroceryItem
            item={item}
            key={`item + ${item._id}`}
          />
        ))
      }
    </div>
    <GroceryListAddItem />
  </div>
);

export default GroceryItemList;