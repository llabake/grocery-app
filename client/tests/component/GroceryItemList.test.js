import React from 'react';
import { shallow } from 'enzyme';

import GroceryItemList from "../../Components/GroceryItemList";
import groceryMock from "../mocks/groceryMock";
import GroceryListAddItem from "../../Components/GroceryListAddItem";

describe('<GroceryItemList />', () => {
  const wrapper = shallow(<GroceryItemList items={groceryMock.groceryItems} />);
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('renders grocery list add item correctly ', () => {
    const tree = shallow(<GroceryItemList items={[groceryMock.createdGroceryWithId]} />);
    expect(tree.find(GroceryListAddItem).length).toBe(1);
  });
});