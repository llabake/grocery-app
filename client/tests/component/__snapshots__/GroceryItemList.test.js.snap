// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroceryItemList /> renders correctly 1`] = `
<div>
  <h1
    className="title"
  >
    Grocery List
  </h1>
  <div>
    <Connect(GroceryItem)
      item={
        Object {
          "name": "mock-grocery-one",
          "purchased": "false",
        }
      }
      key="item + undefined"
    />
    <Connect(GroceryItem)
      item={
        Object {
          "name": "mock-grocery-two",
          "purchased": "true",
        }
      }
      key="item + undefined"
    />
  </div>
  <Connect(GroceryListAddItem) />
</div>
`;
